bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/client/IPClient.java

```diff
@@ -119,11 +189,39 @@ private KNXnetIPConnection getConnection(int serviceMode, @Nullable InetSocketAd
                     }
                 }
                 final InetAddress mcast = remoteEP != null ? remoteEP.getAddress() : null;
-                conn = new KNXnetIPRouting(netIf, mcast);
+                if (ipConnectionType == IpConnectionType.SECURE_ROUTER) {
+                    conn = SecureConnection.newRouting(netIf, mcast, secureRoutingBackboneGroupKey,
+                            Duration.ofMillis(secureRoutingLatencyToleranceMs));
+                } else {
+                    conn = new KNXnetIPRouting(netIf, mcast);
+                }
                 break;
             default:
                 throw new KNXIllegalArgumentException("unknown service mode");
         }
         return conn;
     }
+
+    private void closeTcpConnection() {
+        final SecureSession toBeClosed = tcpSession;
+        if (toBeClosed != null) {
+            tcpSession = null;
+            logger.debug("Bridge {} closing TCP connection", thingUID);
+            try {
+                toBeClosed.close();
+                try {
+                    Thread.sleep(PAUSE_ON_TCP_SESSION_CLOSE_MS);
+                } catch (InterruptedException e) {
+                }
+            } catch (Exception e) {
+                logger.warn("closing TCP connection failed: {}", e.getMessage());
```

Comment:
Not sure about failure cases during close. Calimero cancels some futures, sends connection close packets, etc.
I have the feeling this could happen due to runtime issues. Have not seen it recently, but I don't want to hide it.
-> will change to info

Reply:
You should then reduce it to debug. What is a user supposed to do when he sees that info message? It mainly seems relevant for you and you'll likely be running in debug anyhow. 😉

Reply:
fair enough. debug.